### ISSUE TITLE
Introduce monorepo structure

### ISSUE THREAD

**Issue by kikuomax** [codemonger-io/cdk-rest-api-with-spec#11]
Pros:
- `example` can depend on the latest `cdk-rest-api-with-spec` without a relative path

Cons:
- `npm` may no longer be able to install the package directly from the GitHub repository

**Comment by kikuomax** [comment]
I did some experiment.

1. I moved the library code to `packages/lib` and the example to `packages/example`, configured `workspaces` field of `package.json` and named the root package `cdk-rest-api-with-spec-root`.
2. I pushed the experimental configuration to [my fork](https://github.com/kikuomax/cdk-rest-api-with-spec/tree/monorepo). 
3. I created a new CDK project outside `cdk-rest-api-with-spec`, copied the code from `packages/example` and installed `cdk-rest-api-with-spec-root` from my fork.
4. If I ran `npx cdk synth`, I could not resolve `cdk-rest-api-with-spec`.
5. If I rewrote the import statement as `import { ... } from 'cdk-rest-api-with-spec-root/packages/lib'`, it worked.
